### ISSUE TITLE
`{}` makes decryption results hard to use

### ISSUE THREAD

**Issue by kschaefe** [panva/paseto#35]
### What happened?

With TypeScript strict checking enabled (https://www.typescriptlang.org/tsconfig#strict), `decrypt` returns a type that causes compilation errors.
```
TS7053: Element implicitly has an 'any' type because expression of type '"usr"' can't be used to index type '{}'.
         Property 'usr' does not exist on type '{}'.
```

I would expect that indexed access would work and that the code should compile.

Changing the return from `Promise<object>` to `Promise<Record<PropertyKey, unknown>>` would allow indexed access.

### Version

v3.1.1

### Node.js Version

v16.13.0

### Code to reproduce

```js
paseto.V1.decrypt(token, Buffer.from(key)).then(
    (token) => {
        const uuid = token['usr'];
        // ... do stuff
    }
);
```

### Required

- [X] I have searched the issues tracker and discussions for similar topics and couldn't find anything related.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/panva/paseto/blob/main/CODE_OF_CONDUCT.md)

**Comment by kschaefe** [comment]
Per your guidelines, I have filed this to open a discussion before submitting a PR to fix this problem.

If interested the branch with the failing test for the current code and the fixes is https://github.com/CarnegieLearningWeb/paseto/tree/bugfix/correct-object-types.

**Comment by panva** [comment]
@kschaefe feel free to submit the PR, feedback to be continued in the PR ;)

**Comment by panva** [comment]
Fixed by #36
